torch/functional.py

```diff
@@ -937,15 +937,15 @@ def _consecutive_return_inverse(input, return_inverse=False, return_counts=False
 unique_consecutive.__doc__ = _unique_consecutive_impl.__doc__
 
 
-def tensordot(a, b, dims=2, out=None):
+def tensordot(a, b, dims: Any = 2, out: Optional[torch.Tensor] = None):
```

Comment:
`torch.jit.isinstance` and adding overhead here, we're iterating over the entire list when we use to just check if the input was a list type. I dont think we should be adding overhead to eager operators. 

I think you can support this without making these changes by using overloads:

@overload
def tensordot(a, b, dims: int = 2, out: Optional[torch.Tensor] = None):

@overload
def tensordot(a, b, dims: Tuple[List[int], List[int]], out: Optional[torch.Tensor] = None):

@overload
def tensordot(a, b, dims: List[List[int]], out: Optional[torch.Tensor] = None):

Reply:
Good call! Changed to use `@overload`.